src/derivable/map.ts

```diff
@@ -0,0 +1,83 @@
+import { Derivable, SettableDerivable, State } from '../interfaces';
+import { connect, disconnect, unresolved } from '../symbols';
+import { addObserver, independentTracking, removeObserver } from '../tracking';
+import { ErrorWrapper } from '../utils';
+import { BaseDerivable } from './base-derivable';
+import { BaseDerivation } from './derivation';
+import { isSettableDerivable } from './typeguards';
+
+export class Mapping<B, V> extends BaseDerivation<V> implements Derivable<V> {
+    constructor(
+        protected readonly _base: BaseDerivable<B>,
+        private readonly _pureFn: (state: State<B>) => State<V>,
```

Comment:
It is pretty important that this is a pure function. It creates too much overhead to warn against using a `Derivable.get()` inside a `Mapping` inside a `Derivation`, doesn't it?

What I mean is:
```typescript
const a$ = atom(1);
const m$ = atom(2).map(d => d + a$.get());

m$.react(...); // Won't react to changes in a$;
derive(() => m$.get()).react(...); // Will react to changes in a$ (right?), but needs extra administration to check
```

Reply:
Doesn't work, because of the `independentTracking`, so no warning needed I guess.